pallets/gear/src/queue.rs

```diff
@@ -61,48 +53,91 @@ where
             block_config,
             GasAllowanceOf::<T>::get(),
             dispatch.into_incoming(gas_limit),
-            program_id,
+            destination_id,
         ) {
             Ok(dispatch) => dispatch,
-            Err(journal) => return Ok(journal),
+            Err(journal) => return journal,
         };
 
-        let (precharged_dispatch, actor_data) =
-            get_actor_data(precharged_dispatch).map_err(QueueStepError::ActorData)?;
+        // Can't process messages for non-active program.
+        let Some(program) = Self::get_active_program(destination_id) else {
+            log::trace!("Message is sent to non-active program {:?}", destination_id);
+            return core_processor::process_non_executable(
+                precharged_dispatch,
+                destination_id,
+                ErrorReplyReason::InactiveProgram,
+            );
+        };
+
+        if program.state == ProgramState::Initialized && dispatch_kind == DispatchKind::Init {
+            // Panic is impossible, because gear protocol does not provide functionality
+            // to send second init message to any already existing program.
+            panic!("Init message is sent to already initialized program");
```

Comment:
Actually we don't have to panic here, but maybe better to check this case for future fast fixes

Reply:
Please think carefully about this case, we can always return to error reply here of cause.